sql/wsrep_trans_observer.h

```diff
@@ -303,7 +306,7 @@ static inline void wsrep_after_apply(THD* thd)
 static inline void wsrep_open(THD* thd)
 {
   DBUG_ENTER("wsrep_open");
-  if (wsrep_global_on())
+  if (wsrep_on(thd))
```

Comment:
I think wsrep_on and wsrep_global_on are very different function , wsrep_on is much stricter . But yeah if you decided to use wsrep_on, wont thd->wsrep_cs().state() != wsrep::client_state::s_none be a better choice ?

Reply:
yes, wsrep_on(thd) is stricter, as it looks at wsrep_on value in THD level. Here it is used to see if the session is intending to use wsrep provider for replicating future transactions done in the session. Looking for global wsrep_on value would not be the correct choice here.
wsrep_open() will initialize client state for the session, so client state would not be possible to use yet here.